#! /bin/bash

# FASTQ_1=$1
# FASTQ_2=$2
# SM_TAG=$3
# OUT_DIR=$4
# REF_HASH_DIR=$5
# REF_GENOME=$6
# DBSNP_138='/isilon/sequencing/GATK_resource_bundle/2.8/b37/dbsnp_138.b37.vcf'

# GENERATE VCF and BAM 

for SAMPLE in $(awk 'BEGIN{FS=","} NR>1 {print $8}' /isilon/sequencing/Seq_Proj/H_Sobreira_NCMD__SeqWholeGenome_160531_1/post_joint_call_ss.csv); do dragen -f -r /staging/human/reference/GRC37/human_g1k_v37_decoy_hash_table -1 /isilon/sequencing/Seq_Proj/H_Sobreira_NCMD__SeqWholeGenome_160531_1/FASTQ/$SAMPLE/$SAMPLE"_R1.fastq.gz" -2 /isilon/sequencing/Seq_Proj/H_Sobreira_NCMD__SeqWholeGenome_160531_1/FASTQ/$SAMPLE/$SAMPLE"_R2.fastq.gz" --enable-variant-caller true --vc-sample-name $SAMPLE --vc-reference /staging/human/reference/GRC37/human_g1k_v37_decoy.fasta --output-directory /isilon/sequencing/Seq_Proj/H_Sobreira_NCMD__SeqWholeGenome_160531_1/DRAGEN_OUTPUT/ --intermediate-results-dir /staging/tmp --generate-md-tags true --enable-duplicate-marking false --vc-max-alternate-alleles 3 --enable-map-align-output true --enable-bam-indexing true --RGID $SAMPLE --RGLB $SAMPLE --RGPL ILLUMINA --RGDT '2016-01-16T19:00:00-0500' --RGSM $SAMPLE --RGDS HiSeq-X --output-file-prefix $SAMPLE --dbsnp /isilon/sequencing/GATK_resource_bundle/2.8/b37/dbsnp_138.b37.vcf --enable-vqsr true --vqsr-config-file /opt/edico/config/dragen-VQSR.cfg --vqsr-resource SNP,15.0,/isilon/sequencing/GATK_resource_bundle/2.5/b37/hapmap_3.3.b37.vcf --vqsr-resource SNP,12.0,/isilon/sequencing/GATK_resource_bundle/2.5/b37/1000G_omni2.5.b37.vcf --vqsr-resource SNP,10.0,/isilon/sequencing/GATK_resource_bundle/2.5/b37/1000G_phase1.snps.high_confidence.b37.vcf --vqsr-resource SNP,2.0,/isilon/sequencing/GATK_resource_bundle/2.8/b37/dbsnp_138.b37.excluding_sites_after_129.vcf --vqsr-resource INDEL,12.0,/isilon/sequencing/GATK_resource_bundle/2.2/b37/Mills_and_1000G_gold_standard.indels.b37.vcf --vqsr-tranche 100.0 --vqsr-tranche 99.9 --vqsr-tranche 99.8 --vqsr-tranche 99.7 --vqsr-tranche 99.6 --vqsr-tranche 99.5 --vqsr-tranche 99.4 --vqsr-tranche 99.4 --vqsr-tranche 99.3 --vqsr-tranche 99.2 --vqsr-tranche 99.2 --vqsr-tranche 99.1 --vqsr-tranche 99.1 --vqsr-tranche 99.0 --vqsr-tranche 98.0 --vqsr-tranche 97.0 --vqsr-tranche 96.0 --vqsr-tranche 95.0 --vqsr-tranche 90.0 --vqsr-filter-level SNP,99.9 --vqsr-filter-level INDEL,99.9 --vqsr-lod-cutoff -5.0; done

# GENERATE GVCF

for SAMPLE in $(awk 'BEGIN{FS=","} NR>1 {print $8}' /isilon/sequencing/Seq_Proj/H_Sobreira_NCMD__SeqWholeGenome_160531_1/post_joint_call_ss.csv); do dragen -f -r /staging/human/reference/GRC37/human_g1k_v37_decoy_hash_table -1 /isilon/sequencing/Seq_Proj/H_Sobreira_NCMD__SeqWholeGenome_160531_1/FASTQ/$SAMPLE/$SAMPLE"_R1.fastq.gz" -2 /isilon/sequencing/Seq_Proj/H_Sobreira_NCMD__SeqWholeGenome_160531_1/FASTQ/$SAMPLE/$SAMPLE"_R2.fastq.gz" --enable-variant-caller true --vc-sample-name $SAMPLE --vc-reference /staging/human/reference/GRC37/human_g1k_v37_decoy.fasta --output-directory /isilon/sequencing/Seq_Proj/H_Sobreira_NCMD__SeqWholeGenome_160531_1/DRAGEN_OUTPUT/ --intermediate-results-dir /staging/tmp --generate-md-tags true --enable-duplicate-marking false --vc-max-alternate-alleles 3 --enable-map-align-output true --enable-bam-indexing true --RGID $SAMPLE --RGLB $SAMPLE --RGPL ILLUMINA --RGDT '2016-01-16T19:00:00-0500' --RGSM $SAMPLE --RGDS HiSeq-X --output-file-prefix $SAMPLE --dbsnp /isilon/sequencing/GATK_resource_bundle/2.8/b37/dbsnp_138.b37.vcf --enable-vqsr true --vqsr-config-file /opt/edico/config/dragen-VQSR.cfg --vqsr-resource SNP,15.0,/isilon/sequencing/GATK_resource_bundle/2.5/b37/hapmap_3.3.b37.vcf --vqsr-resource SNP,12.0,/isilon/sequencing/GATK_resource_bundle/2.5/b37/1000G_omni2.5.b37.vcf --vqsr-resource SNP,10.0,/isilon/sequencing/GATK_resource_bundle/2.5/b37/1000G_phase1.snps.high_confidence.b37.vcf --vqsr-resource SNP,2.0,/isilon/sequencing/GATK_resource_bundle/2.8/b37/dbsnp_138.b37.excluding_sites_after_129.vcf --vqsr-resource INDEL,12.0,/isilon/sequencing/GATK_resource_bundle/2.2/b37/Mills_and_1000G_gold_standard.indels.b37.vcf --vqsr-tranche 100.0 --vqsr-tranche 99.9 --vqsr-tranche 99.8 --vqsr-tranche 99.7 --vqsr-tranche 99.6 --vqsr-tranche 99.5 --vqsr-tranche 99.4 --vqsr-tranche 99.4 --vqsr-tranche 99.3 --vqsr-tranche 99.2 --vqsr-tranche 99.2 --vqsr-tranche 99.1 --vqsr-tranche 99.1 --vqsr-tranche 99.0 --vqsr-tranche 98.0 --vqsr-tranche 97.0 --vqsr-tranche 96.0 --vqsr-tranche 95.0 --vqsr-tranche 90.0 --vqsr-filter-level SNP,99.9 --vqsr-filter-level INDEL,99.9 --vqsr-lod-cutoff -5.0 --vc-emit-ref-confidence GVCF ; done

# dragen -f -r $REF_HASH_DIR -1 $FASTQ_1 -2 $FASTQ_2 --enable-variant-caller true --vc-sample-name $SM_TAG  --vc-reference $REF_GENOME --output-directory $OUT_DIR --intermediate-results-dir /staging/tmp --generate-md-tags true --enable-duplicate-marking false --vc-max-alternate-alleles 3 --combine-samples-by-name=true --output-file-prefix $SM_TAG --vc-emit-ref-confidence GVCF --dbsnp $DBSNP_138 --enable-map-align-output true --enable-bam-indexing true
