#$ -S /bin/bash
#$ -q rnd.q,test.q,prod.q
#$ -cwd
#$ -V
#$ -p -1000

set

JAVA_1_7=$1
GATK_DIR=$2
KEY=$3
REF_GENOME=$4
CORE_PATH=$5
PROJECT=$6
SM_TAG=$7

RIS_ID=${SM_TAG%@*}
BARCODE_2D=${SM_TAG#*@}

START_ANNOTATE_VCF=`date '+%s'`

# ANNOTATE THE VCF

$JAVA_1_7/java -jar $GATK_DIR/GenomeAnalysisTK.jar \
-T VariantAnnotator \
-R $REF_GENOME \
--variant $CORE_PATH/$PROJECT'/DRAGEN_OUTPUT/'$SM_TAG'.hard-filtered.vcf' \
-L 1 -L 2 -L 3 -L 4 -L 5 -L 6 -L 7 -L 8 -L 9 -L 10 \
-L 11 -L 12 -L 13 -L 14 -L 15 -L 16 -L 17 -L 18 -L 19 -L 20 -L 21 -L 22 \
-L X -L Y -L MT \
-A GCContent \
-A VariantType \
--disable_auto_index_creation_and_locking_when_reading_rods \
-et NO_ET \
-K $KEY \
-o $CORE_PATH/$PROJECT/VCF/SINGLE/WHOLE_GENOME/$SM_TAG".WHOLE.GENOME.vcf"

END_ANNOTATE_VCF=`date '+%s'`

echo 'ANNOTATE_VCF,B.01,'$START_ANNOTATE_VCF','$END_ANNOTATE_VCF >> $CORE_PATH/$PROJECT/REPORTS/run_times.csv

# GenotypeSummaries apparently needs a version higher than 3.1-1

# $JAVA_1_7/java -jar $GATK_DIR/GenomeAnalysisTK.jar \
# -T VariantAnnotator \
# -R $REF_GENOME \
# --variant $CORE_PATH/$PROJECT/$DNA_HASH_ADDRESS/$MDNA_HASH_ADDRESS/VQSR/variants.all.vcf \
# -L $CORE_PATH/$PROJECT/$DNA_HASH_ADDRESS/$MDNA_HASH_ADDRESS/VQSR/variants.all.vcf \
# -A GenotypeSummaries \
# -A SampleList \
# -A GCContent \
# -A VariantType \
# --disable_auto_index_creation_and_locking_when_reading_rods \
# -et NO_ET \
# -K $KEY \
# -o $CORE_PATH/$PROJECT/VCF/SINGLE/WHOLE_GENOME/$SM_TAG".WHOLE.GENOME.vcf"

# qsub -N "B.01.A.01_INDEX_VCF_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".INDEX.VCF.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".INDEX.VCF.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.01_INDEX_VCF.sh $1 $2 $3 $4
# 
# sleep 3s
# 
# qsub -N "B.01.A.02_EXTRACT_VCF_CODING_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.VCF.CODING.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.VCF.CODING.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.02_EXTRACT_VCF_CODING.sh $1 $2 $3 $4
# 
# sleep 3s
# 
# qsub -N "B.01.A.03_EXTRACT_VCF_WHOLE_GENOME_PASS_VARIANT_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.VCF.WHOLE.GENOME.PASS.VARIANT.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.VCF.WHOLE.GENOME.PASS.VARIANT.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.03_EXTRACT_VCF_WHOLE_GENOME_PASS_VARIANT.sh $1 $2 $3 $4
# 
# sleep 3s
# 
# qsub -N "B.01.A.04_EXTRACT_VCF_CODING_PASS_VARIANT_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.VCF.CODING.PASS.VARAINT.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.VCF.CODING.PASS.VARIANT.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.04_EXTRACT_VCF_CODING_PASS_VARIANT.sh $1 $2 $3 $4
# 
# sleep 3s
# 
# qsub -N "B.01.A.05_EXTRACT_SNV_WHOLE_GENOME_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.SNV.WHOLE.GENOME.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.SNV.WHOLE.GENOME.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.05_EXTRACT_SNV_WHOLE_GENOME.sh $1 $2 $3 $4
# 
# sleep 3s
# 
# qsub -N "B.01.A.06_EXTRACT_SNV_CODING_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.SNV.CODING.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.SNV.CODING.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.06_EXTRACT_SNV_CODING.sh $1 $2 $3 $4
# sleep 3s
# 
# qsub -N "B.01.A.07_EXTRACT_SNV_WHOLE_GENOME_PASS_VARIANT_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.SNV.WHOLE.GENOME.PASS.VARIANT.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.SNV.WHOLE.GENOME.PASS.VARIANT.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.07_EXTRACT_SNV_WHOLE_GENOME_PASS_VARIANT.sh $1 $2 $3 $4
# 
# sleep 3s
# 
# qsub -N "B.01.A.08_EXTRACT_SNV_CODING_PASS_VARIANT_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.SNV.CODING.PASS.VARAINT.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.SNV.CODING.PASS.VARIANT.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.08_EXTRACT_SNV_CODING_PASS_VARIANT.sh $1 $2 $3 $4
# sleep 3s
# 
# qsub -N "B.01.A.09_EXTRACT_INDEL_WHOLE_GENOME_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.INDEL.WHOLE.GENOME.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.INDEL.WHOLE.GENOME.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.09_EXTRACT_INDEL_WHOLE_GENOME.sh $1 $2 $3 $4
# 
# sleep 3s
# 
# qsub -N "B.01.A.10_EXTRACT_INDEL_CODING_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.INDEL.CODING.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.INDEL.CODING.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.10_EXTRACT_INDEL_CODING.sh $1 $2 $3 $4
# sleep 3s
# 
# qsub -N "B.01.A.11_EXTRACT_INDEL_WHOLE_GENOME_PASS_VARIANT_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.INDEL.WHOLE.GENOME.PASS.VARIANT.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.INDEL.WHOLE.GENOME.PASS.VARIANT.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.11_EXTRACT_INDEL_WHOLE_GENOME_PASS_VARIANT.sh $1 $2 $3 $4
# 
# sleep 3s
# 
# qsub -N "B.01.A.12_EXTRACT_INDEL_CODING_PASS_VARIANT_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.INDEL.CODING.PASS.VARAINT.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.INDEL.CODING.PASS.VARIANT.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.12_EXTRACT_INDEL_CODING_PASS_VARIANT.sh $1 $2 $3 $4
