#$ -S /bin/bash
#$ -q rnd.q,test.q,prod.q
#$ -cwd
#$ -V
#$ -p -1000

set

JAVA_1_7=$1
GATK_DIR=$2
KEY=$3
CODING_BED=$4
CODING_BED_MT=$5
REF_GENOME=$6
CORE_PATH=$7
PROJECT=$8
SM_TAG=$9

RIS_ID=${SM_TAG%@*}
BARCODE_2D=${SM_TAG#*@}


START_EXTRACT_SNV_CODING_PASS=`date '+%s'`

# Filter snv only, REMOVE NON-VARIANT, FAIL, CODING

$JAVA_1_7/java -jar $GATK_DIR/GenomeAnalysisTK.jar \
-T SelectVariants \
-R $REF_GENOME \
-sn $SM_TAG \
-ef \
-env \
--keepOriginalAC \
-selectType SNP \
-L $CODING_BED \
-L $CODING_BED_MT \
-et NO_ET \
-K $KEY \
--variant $CORE_PATH/$PROJECT/VCF/SINGLE/WHOLE_GENOME/$SM_TAG".WHOLE.GENOME.vcf" \
-o $CORE_PATH/$PROJECT/SNV/SINGLE/CODING/PASS/$SM_TAG".CODING.SNV.PASS.vcf"

END_EXTRACT_SNV_CODING_PASS=`date '+%s'`

echo 'EXTRACT_SNV_CODING_PASS,B.01.A.08,'$START_EXTRACT_SNV_CODING_PASS','$END_EXTRACT_SNV_CODING_PASS >> $CORE_PATH/$PROJECT/REPORTS/run_times.csv


# qsub -N "B.01.A.08.A.01_TI_TV_CODING_PASS_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".TITV.CODING.PASS.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".TITV.CODING.PASS.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.08.A.01_TI_TV_CODING_PASS.sh $1 $2 $3 $4
# 
# sleep 3s
# 
# qsub -N "B.01.A.08.A.02_TI_TV_CODING_KNOWN_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".TITV.CODING.KNOWN.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".TITV.CODING.KNOWN.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.08.A.02_TI_TV_CODING_KNOWN.sh $1 $2 $3 $4
# 
# sleep 3s
# 
# qsub -N "B.01.A.08.A.03_TI_TV_CODING_NOVEL_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".TITV.CODING.NOVEL.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".TITV.CODING.NOVEL.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.08.A.03_TI_TV_CODING_NOVEL.sh $1 $2 $3 $4 $5 $6
