#$ -S /bin/bash
#$ -q rnd.q,test.q,prod.q
#$ -cwd
#$ -V
#$ -p -1000

set

JAVA_1_7=$1
GATK_DIR=$2
KEY=$3
REF_GENOME=$4
CORE_PATH=$5
PROJECT=$6
SM_TAG=$7

RIS_ID=${SM_TAG%@*}
BARCODE_2D=${SM_TAG#*@}

START_EXTRACT_WG_SNV_PASS=`date '+%s'`

# Filter snv only, REMOVE NON-VARIANT, FAIL

$JAVA_1_7/java -jar $GATK_DIR/GenomeAnalysisTK.jar \
-T SelectVariants \
-R $REF_GENOME \
-sn $SM_TAG \
-ef \
-env \
--keepOriginalAC \
-selectType SNP \
-et NO_ET \
-K $KEY \
--variant $CORE_PATH/$PROJECT/VCF/SINGLE/WHOLE_GENOME/$SM_TAG".WHOLE.GENOME.vcf" \
-o $CORE_PATH/$PROJECT/SNV/SINGLE/WHOLE_GENOME/PASS/$SM_TAG".WHOLE.GENOME.SNV.PASS.vcf"

END_EXTRACT_WG_SNV_PASS=`date '+%s'`

echo 'EXTRACT_WG_SNV_PASS,B.01.A.07,'$START_EXTRACT_WG_SNV_PASS','$END_EXTRACT_WG_SNV_PASS >> $CORE_PATH/$PROJECT/REPORTS/run_times.csv

# qsub -N "B.01.A.07.A.01_TI_TV_WHOLE_GENOME_PASS_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".TITV.WHOLE.GENOME.PASS.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".TITV.WHOLE.GENOME.PASS.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.07.A.01_TI_TV_WHOLE_GENOME_PASS.sh $1 $2 $3 $4
# 
# sleep 3s
# 
# qsub -N "B.01.A.07.A.02_TI_TV_WHOLE_GENOME_KNOWN_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".TITV.WHOLE.GENOME.KNOWN.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".TITV.WHOLE.GENOME.KNOWN.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.07.A.02_TI_TV_WHOLE_GENOME_KNOWN.sh $1 $2 $3 $4
# 
# sleep 3s
# 
# qsub -N "B.01.A.07.A.03_TI_TV_WHOLE_GENOME_NOVEL_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".TITV.WHOLE.GENOME.NOVEL.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".TITV.WHOLE.GENOME.NOVEL.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.07.A.03_TI_TV_WHOLE_GENOME_NOVEL.sh $1 $2 $3 $4
