#$ -S /bin/bash
#$ -q rnd.q,test.q,prod.q
#$ -cwd
#$ -V
#$ -p -1000

set

PROJECT=$1
SM_TAG=$2
REF_GENOME=$3
DBSNP=$4

RIS_ID=${SM_TAG%@*}
BARCODE_2D=${SM_TAG#*@}

JAVA_1_7="/isilon/sequencing/Kurt/Programs/Java/jdk1.7.0_25/bin"
CORE_PATH="/isilon/sequencing/Seq_Proj"
BWA_DIR="/isilon/sequencing/Kurt/Programs/BWA/bwa-0.7.8/"
PICARD_DIR="/isilon/sequencing/Kurt/Programs/Picard/picard-tools-1.118"
GATK_DIR="/isilon/sequencing/CIDRSeqSuiteSoftware/gatk/GATK_3/GenomeAnalysisTK-3.1-1"
VERIFY_DIR="/isilon/sequencing/Kurt/Programs/VerifyBamID/verifyBamID_20120620/bin/"
BED_DIR=$CORE_PATH"/BED_Files/"
GENE_LIST="/isilon/sequencing/CIDRSeqSuiteSoftware/RELEASES/5.0.0/aux_files/RefSeqGene.GRCh37.Ready.txt"
VERIFY_VCF="/isilon/sequencing/CIDRSeqSuiteSoftware/RELEASES/5.0.0/aux_files/Omni25_genotypes_1525_samples_v2.b37.PASS.ALL.sites.vcf"
CODING_BED="/isilon/sequencing/CIDRSeqSuiteSoftware/RELEASES/5.0.0/aux_files/UCSC_hg19_CodingOnly_083013_MERGED_noContigs_noCHR.bed"
CODING_BED_MT="/isilon/sequencing/CIDRSeqSuiteSoftware/RELEASES/5.0.0/aux_files/MT.coding.bed"
TRANSCRIPT_BED="/isilon/sequencing/CIDRSeqSuiteSoftware/RELEASES/5.0.0/aux_files/Transcripts.UCSC.Merged.NoContigsAlts.bed"
TRANSCRIPT_BED_MT="/isilon/sequencing/CIDRSeqSuiteSoftware/RELEASES/5.0.0/aux_files/MT.transcripts.bed"
GAP_BED="/isilon/sequencing/CIDRSeqSuiteSoftware/RELEASES/5.0.0/aux_files/GRCh37.gaps.bed"
SAMTOOLS_DIR="/isilon/sequencing/Kurt/Programs/samtools/samtools-0.1.18/"
TABIX_DIR="/isilon/sequencing/Kurt/Programs/TABIX/tabix-0.2.6/"
LUMPY_DIR="/isilon/sequencing/Kurt/Programs/LUMPY/lumpy-sv-master"
KEY="/isilon/sequencing/CIDRSeqSuiteSoftware/gatk/GATK_2/lee.watkins_jhmi.edu.key"

mkdir -p $CORE_PATH/$PROJECT/LOGS
mkdir -p $CORE_PATH/$PROJECT/BAM/{AGGREGATE,READGROUP}
mkdir -p $CORE_PATH/$PROJECT/HC_BAM
mkdir -p $CORE_PATH/$PROJECT/GVCF
mkdir -p $CORE_PATH/$PROJECT/REPORTS/{ALIGNMENT_SUMMARY,ANNOVAR,PICARD_DUPLICATES,QC_REPORTS,TI_TV,TI_TV_MS,VERIFY_BAM_ID,SAMPLE_SHEETS,OXIDATION,JUMPING,ESTIMATE_LIBRARY}
mkdir -p $CORE_PATH/$PROJECT/REPORTS/TI_TV/{WHOLE_GENOME,CODING}
mkdir -p $CORE_PATH/$PROJECT/REPORTS/TI_TV_MS/{WHOLE_GENOME,CODING}
mkdir -p $CORE_PATH/$PROJECT/REPORTS/BASECALL_Q_SCORE_DISTRIBUTION/{METRICS,PDF}
mkdir -p $CORE_PATH/$PROJECT/REPORTS/CONCORDANCE/{WHOLE_GENOME,CODING}
mkdir -p $CORE_PATH/$PROJECT/REPORTS/CONCORDANCE_MS/{WHOLE_GENOME,CODING}
mkdir -p $CORE_PATH/$PROJECT/REPORTS/COUNT_COVARIATES/{GATK_REPORT,PDF}
mkdir -p $CORE_PATH/$PROJECT/REPORTS/GC_BIAS/{METRICS,PDF,SUMMARY}
mkdir -p $CORE_PATH/$PROJECT/REPORTS/INSERT_SIZE/{METRICS,PDF}
mkdir -p $CORE_PATH/$PROJECT/REPORTS/LOCAL_REALIGNMENT_INTERVALS
mkdir -p $CORE_PATH/$PROJECT/REPORTS/MEAN_QUALITY_BY_CYCLE/{METRICS,PDF}
mkdir -p $CORE_PATH/$PROJECT/REPORTS/DEPTH_OF_COVERAGE/{DEPTH_SUMMARY,CODING_COVERAGE,TRANSCRIPT_COVERAGE}
mkdir -p $CORE_PATH/$PROJECT/TEMP
mkdir -p $CORE_PATH/$PROJECT/FASTQ
mkdir -p $CORE_PATH/$PROJECT/BED_Files
mkdir -p $CORE_PATH/$PROJECT/VCF/SINGLE/WHOLE_GENOME/PASS
mkdir -p $CORE_PATH/$PROJECT/VCF/SINGLE/CODING/PASS
mkdir -p $CORE_PATH/$PROJECT/VCF/MULTI/WHOLE_GENOME/{PASS_ALL,PASS_VARIANT}
mkdir -p $CORE_PATH/$PROJECT/VCF/MULTI/CODING/{PASS_ALL,PASS_VARIANT}
mkdir -p $CORE_PATH/$PROJECT/SNV/SINGLE/WHOLE_GENOME/PASS
mkdir -p $CORE_PATH/$PROJECT/SNV/SINGLE/CODING/PASS
mkdir -p $CORE_PATH/$PROJECT/SNV/MULTI/WHOLE_GENOME/{PASS_ALL,PASS_VARIANT}
mkdir -p $CORE_PATH/$PROJECT/SNV/MULTI/CODING/{PASS_ALL,PASS_VARIANT}
mkdir -p $CORE_PATH/$PROJECT/INDEL/SINGLE/WHOLE_GENOME/PASS
mkdir -p $CORE_PATH/$PROJECT/INDEL/SINGLE/CODING/PASS
mkdir -p $CORE_PATH/$PROJECT/INDEL/MULTI/WHOLE_GENOME/{PASS_ALL,PASS_VARIANT}
mkdir -p $CORE_PATH/$PROJECT/INDEL/MULTI/CODING/{PASS_ALL,PASS_VARIANT}
mkdir -p $CORE_PATH/$PROJECT/LUMPY/{RAW,UCSC_CODING}

START_ANNOTATE_VCF=`date '+%s'`

# ANNOTATE THE VCF

$JAVA_1_7/java -jar $GATK_DIR/GenomeAnalysisTK.jar \
-T VariantAnnotator \
-R $REF_GENOME \
--variant $CORE_PATH/$PROJECT'/DRAGEN_OUTPUT/'$SM_TAG'.vqsr.vcf' \
-L 1 -L 2 -L 3 -L 4 -L 5 -L 6 -L 7 -L 8 -L 9 -L 10 \
-L 11 -L 12 -L 13 -L 14 -L 15 -L 16 -L 17 -L 18 -L 19 -L 20 -L 21 -L 22 \
-L X -L Y -L MT \
--dbsnp $DBSNP \
-A GCContent \
-A VariantType \
--disable_auto_index_creation_and_locking_when_reading_rods \
-et NO_ET \
-K $KEY \
-o $CORE_PATH/$PROJECT/VCF/SINGLE/WHOLE_GENOME/$SM_TAG".WHOLE.GENOME.vcf"

END_ANNOTATE_VCF=`date '+%s'`

echo 'ANNOTATE_VCF,B.01,'$START_ANNOTATE_VCF','$END_ANNOTATE_VCF >> $CORE_PATH/$PROJECT/REPORTS/run_times.csv

# --variant $CORE_PATH/$PROJECT'/DRAGEN_OUTPUT/'$SM_TAG'.vqsr.vcf' \

# GenotypeSummaries apparently needs a version higher than 3.1-1

# $JAVA_1_7/java -jar $GATK_DIR/GenomeAnalysisTK.jar \
# -T VariantAnnotator \
# -R $REF_GENOME \
# --variant $CORE_PATH/$PROJECT/$DNA_HASH_ADDRESS/$MDNA_HASH_ADDRESS/VQSR/variants.all.vcf \
# -L $CORE_PATH/$PROJECT/$DNA_HASH_ADDRESS/$MDNA_HASH_ADDRESS/VQSR/variants.all.vcf \
# -A GenotypeSummaries \
# -A SampleList \
# -A GCContent \
# -A VariantType \
# --disable_auto_index_creation_and_locking_when_reading_rods \
# -et NO_ET \
# -K $KEY \
# -o $CORE_PATH/$PROJECT/VCF/SINGLE/WHOLE_GENOME/$SM_TAG".WHOLE.GENOME.vcf"

# qsub -N "B.01.A.01_INDEX_VCF_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".INDEX.VCF.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".INDEX.VCF.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.01_INDEX_VCF.sh $1 $2 $3 $4
# 
# sleep 3s
# 
# qsub -N "B.01.A.02_EXTRACT_VCF_CODING_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.VCF.CODING.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.VCF.CODING.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.02_EXTRACT_VCF_CODING.sh $1 $2 $3 $4
# 
# sleep 3s
# 
# qsub -N "B.01.A.03_EXTRACT_VCF_WHOLE_GENOME_PASS_VARIANT_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.VCF.WHOLE.GENOME.PASS.VARIANT.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.VCF.WHOLE.GENOME.PASS.VARIANT.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.03_EXTRACT_VCF_WHOLE_GENOME_PASS_VARIANT.sh $1 $2 $3 $4
# 
# sleep 3s
# 
# qsub -N "B.01.A.04_EXTRACT_VCF_CODING_PASS_VARIANT_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.VCF.CODING.PASS.VARAINT.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.VCF.CODING.PASS.VARIANT.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.04_EXTRACT_VCF_CODING_PASS_VARIANT.sh $1 $2 $3 $4
# 
# sleep 3s
# 
# qsub -N "B.01.A.05_EXTRACT_SNV_WHOLE_GENOME_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.SNV.WHOLE.GENOME.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.SNV.WHOLE.GENOME.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.05_EXTRACT_SNV_WHOLE_GENOME.sh $1 $2 $3 $4
# 
# sleep 3s
# 
# qsub -N "B.01.A.06_EXTRACT_SNV_CODING_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.SNV.CODING.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.SNV.CODING.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.06_EXTRACT_SNV_CODING.sh $1 $2 $3 $4
# sleep 3s
# 
# qsub -N "B.01.A.07_EXTRACT_SNV_WHOLE_GENOME_PASS_VARIANT_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.SNV.WHOLE.GENOME.PASS.VARIANT.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.SNV.WHOLE.GENOME.PASS.VARIANT.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.07_EXTRACT_SNV_WHOLE_GENOME_PASS_VARIANT.sh $1 $2 $3 $4
# 
# sleep 3s
# 
# qsub -N "B.01.A.08_EXTRACT_SNV_CODING_PASS_VARIANT_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.SNV.CODING.PASS.VARAINT.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.SNV.CODING.PASS.VARIANT.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.08_EXTRACT_SNV_CODING_PASS_VARIANT.sh $1 $2 $3 $4
# sleep 3s
# 
# qsub -N "B.01.A.09_EXTRACT_INDEL_WHOLE_GENOME_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.INDEL.WHOLE.GENOME.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.INDEL.WHOLE.GENOME.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.09_EXTRACT_INDEL_WHOLE_GENOME.sh $1 $2 $3 $4
# 
# sleep 3s
# 
# qsub -N "B.01.A.10_EXTRACT_INDEL_CODING_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.INDEL.CODING.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.INDEL.CODING.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.10_EXTRACT_INDEL_CODING.sh $1 $2 $3 $4
# sleep 3s
# 
# qsub -N "B.01.A.11_EXTRACT_INDEL_WHOLE_GENOME_PASS_VARIANT_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.INDEL.WHOLE.GENOME.PASS.VARIANT.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.INDEL.WHOLE.GENOME.PASS.VARIANT.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.11_EXTRACT_INDEL_WHOLE_GENOME_PASS_VARIANT.sh $1 $2 $3 $4
# 
# sleep 3s
# 
# qsub -N "B.01.A.12_EXTRACT_INDEL_CODING_PASS_VARIANT_"$RIS_ID"_"$BARCODE_2D \
# -o $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.INDEL.CODING.PASS.VARAINT.SINGLE.log" \
# -e $CORE_PATH/$PROJECT/LOGS/$SM_TAG".EXTRACT.INDEL.CODING.PASS.VARIANT.SINGLE.log" \
# $CORE_PATH/$PROJECT/SCRIPTS/B.01.A.12_EXTRACT_INDEL_CODING_PASS_VARIANT.sh $1 $2 $3 $4
